src/Microsoft.ML.Transforms/TensorFlow/Tensor.cs

```diff
@@ -0,0 +1,919 @@
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Int32, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (int), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a boolean value.
+		/// </summary>
+		public unsafe TFTensor (bool value)
+		{
+			var v = (bool*)Marshal.AllocHGlobal (sizeof (bool));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Bool, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (int), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from an sbyte value.
+		/// </summary>
+		public unsafe TFTensor (sbyte value)
+		{
+			var v = (sbyte*)Marshal.AllocHGlobal (sizeof (sbyte));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Int8, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (sbyte), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a short value.
+		/// </summary>
+		public unsafe TFTensor (short value)
+		{
+			var v = (short*)Marshal.AllocHGlobal (sizeof (short));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Int16, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (short), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from an ushort value.
+		/// </summary>
+		public unsafe TFTensor (ushort value)
+		{
+			var v = (ushort*)Marshal.AllocHGlobal (sizeof (ushort));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Int16, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (ushort), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from an byte value.
+		/// </summary>
+		public unsafe TFTensor (byte value)
+		{
+			var v = (int*)Marshal.AllocHGlobal (sizeof (byte));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.UInt8, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (byte), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a Complex value.
+		/// </summary>
+		public unsafe TFTensor (Complex value)
+		{
+			var v = (Complex*)Marshal.AllocHGlobal (sizeof (Complex));
+			*v = value;
+			handle = TF_NewTensor (TFDataType.Complex128, zeroDims: IntPtr.Zero, num_dims: 0, data: (IntPtr)v, len: (UIntPtr)sizeof (Complex), deallocator: FreeTensorDataDelegate, deallocator_arg: IntPtr.Zero);
+		}
+
+		/// <summary>
+		/// Creates a constant tensor with a single dimension from a float value.
+		/// </summary>
+		public unsafe TFTensor (float value)
```

Comment:
>public unsafe TFTensor (float value) [](start = 2, length = 36)

Can we have a templated versions of creating TFTensor from type <T>?

Reply:
We can have generic factory method that does that.  It'll need to handle T's specifically via switch.  What Ts do you expect it to handle?  Scalars, Arrays of scalars, both?

---
In reply to: [210351058](https://github.com/zeahmed/machinelearning/pull/4#discussion_r210351058) [](ancestors = 210351058)

Reply:
For scalar only right now. Looks like SetupTensor method can work for arrays as it's already being used for arrays underneath every constructor.

---
In reply to: [210355494](https://github.com/zeahmed/machinelearning/pull/4#discussion_r210355494) [](ancestors = 210355494,210351058)

Reply:
Yes, SetupTensor can handle arrays and does the right thing pinning the managed array to avoid a copy.  Note that will only work if the caller knows they have an array, if the caller doesn't and itself is a generic on T that might be an array we'll need to make the factory method handle it to.

---
In reply to: [210356670](https://github.com/zeahmed/machinelearning/pull/4#discussion_r210356670) [](ancestors = 210356670,210355494,210351058)

Reply:
I can use this method directly. This method takes TFDataType as input and I have that information from model. I can use this method for arrays. For scalar there is no such method so I needed a templated one.

---
In reply to: [210360411](https://github.com/zeahmed/machinelearning/pull/4#discussion_r210360411) [](ancestors = 210360411,210356670,210355494,210351058)

Reply:
Yeah, I'll make something, you may want to use it in both scenarios.  Stay tuned.

---
In reply to: [210360411](https://github.com/zeahmed/machinelearning/pull/4#discussion_r210360411) [](ancestors = 210360411,210356670,210355494,210351058)